src/states/common/base.rs

```diff
@@ -333,27 +332,17 @@ pub trait Base: Display {
             );
         }
 
-        // initially only send to dg_size targets
-        for conn_info in conn_infos.iter().take(dg_size) {
-            self.send_message_over_network(conn_info.clone(), &message, token);
-        }
-
-        self.network_service_mut()
-            .targets_cache_mut()
-            .insert_message(token, conn_infos, dg_size);
+        self.send_message_to_initial_targets(conn_infos, dg_size, message);
     }
 
-    fn send_message_over_network(
+    fn send_message_to_initial_targets(
         &mut self,
-        conn_info: ConnectionInfo,
-        message: &Message,
-        token: Token,
+        conn_infos: Vec<ConnectionInfo>,
+        dg_size: usize,
+        message: Message,
     ) {
-        match to_network_bytes(message) {
-            Ok(bytes) => self
-                .network_service_mut()
-                .service_mut()
-                .send(conn_info, bytes, token),
+        let bytes = match to_network_bytes(&message) {
```

Comment:
wondering why not use the previous pattern:
`match () {
  Ok() => send,
  Err() => comment
}`
Which is bit cleaner ?

Reply:
I was finding it a bit confusing.
This function is about sending thing, but the sending was nested making it not the focal point of the function.

Now there is a clear separation: First get bytes, If no failure proceed with sending.


It follows the same control structure as something like:
```
let bytes = make_the_bytes()?;
send_the_bytes(bytes);
```